### ISSUE TITLE
Check Win

### ISSUE THREAD

**Issue by sampyz** [Lu-Yi-Hsun/iqoptionapi#14]
Hello,

is it possible to edit api:
I_want_money.check_win() like I_want_money.check_win("EURUSD")?

because i need to check win in a multiple bet

Thanks in advance!

**Comment by Lu-Yi-Hsun** [comment]
@sampyz  ok will try to implement that today

**Comment by sampyz** [comment]
@Lu-Yi-Hsun i love you!

**Comment by Lu-Yi-Hsun** [comment]
@sampyz 
try get last version and try this code
```
from iqoptionapi.stable_api import IQ_Option
import logging
import time
logging.basicConfig(level=logging.DEBUG,format='%(asctime)s %(message)s')
I_want_money=IQ_Option("email","password")

ok,id_number=I_want_money.buy(2,"EURUSD","call",1,force_buy=False)
print(id_number)
ans=I_want_money.check_win(id_number)
print(ans)
```

**Comment by sampyz** [comment]
@Lu-Yi-Hsun Thanks!
I will try tomorrow!!!
If my bots works i will pay for your job! ;)

**Comment by sampyz** [comment]
@Lu-Yi-Hsun 
hi man, i have try but doesnt work for me...
I_want_money.check_win(id_number) is an infinte loop without results

**Comment by sampyz** [comment]
@Lu-Yi-Hsun 
hello,
I have a problem when i try to send id_number to another script running I_want_money.check_win trought a .ini File..

What is it "ok," before id_number=I_want_money.buy(2,"EURUSD","call",1,force_buy=False)?
Thanks for your helping

**Comment by Lu-Yi-Hsun** [comment]
"ok"  return if "buy" ok(True/False) 
some time network or some problem buy not sucess it will return False

```
ok,id_number=I_want_money.buy(2,"EURUSD","call",1,force_buy=True)
```
if you force_buy set true it will try buy until scess

**Comment by sampyz** [comment]
Thanks for your assistance!!
Do you know if is it possible to checkwin 2 or more bets at the same time or in the same script??

**Comment by Lu-Yi-Hsun** [comment]
#sampyz
You can try thread

**Comment by sampyz** [comment]
@Lu-Yi-Hsun 
Hi mister, i tryed check_win_v2 but is a little bit slowly...i have the result after 5 second...is it possible to make it faster?

Thanks!!!

**Comment by Lu-Yi-Hsun** [comment]
@sampyz 
i set sleep 5 sec  to Polling check if win,so it is slowly.
i will change to 0.5 sec i thing it will better.
you can try last version

**Comment by jackynguyen93** [comment]
@Lu-Yi-Hsun I got an issue when enter multiple order at a time. 
ex: 
id_number1=I_want_money.buy(2,"EURUSD","call",1,force_buy=False)
id_number2=I_want_money.buy(2,"EURJPY","call",1,force_buy=False)
id_number3=I_want_money.buy(2,"EURAUD","call",1,force_buy=False)

The problem is id_number1 = id_number2 = id_number3, all has the same value. So I cannot get the result. 
Note that I am using multiple threads

**Comment by Lu-Yi-Hsun** [comment]
@jackynguyen93

Ok I will fix that

**Comment by jackynguyen93** [comment]
@Lu-Yi-Hsun I really appreciate that. Thanks so much

**Comment by Lu-Yi-Hsun** [comment]
@jackynguyen93  

v 3.6.3 fix

**Comment by jackynguyen93** [comment]
@Lu-Yi-Hsun It seems still happen
My application run 2 threads at the same time but both threads response the same ID
![Untitled](https://user-images.githubusercontent.com/14976555/55395206-4c364f00-556b-11e9-8f1c-36debaceb856.png)
You can see my log show 5211822983 is the response of 2 buy difference requests

**Comment by Lu-Yi-Hsun** [comment]
Not thread safe, you need to use multi connect

**Comment by jackynguyen93** [comment]
@Lu-Yi-Hsun could you explain more. Do I need to have a connect like _I_want_money=IQ_Option("email","password")_  everytime I enter an order?

**Comment by Lu-Yi-Hsun** [comment]
@jackynguyen93  if you want thread ,you need to connect other

**Comment by jackynguyen93** [comment]
So, is it correct if in my order function I put **new_connect=IQ_Option("email","password")**  and use **new_connect** to enter the order ?

**Comment by Lu-Yi-Hsun** [comment]
Yes you need to build other connect ,I suggest use multiprocessing for multi cpu

**Comment by jackynguyen93** [comment]
Could you provide me an example or some article related to this? Forgive for my lack of knowledge, python is not my main language

**Comment by Lu-Yi-Hsun** [comment]
```python

from iqoptionapi.stable_api import IQ_Option
import multiprocessing as mp
import time
import logging
import time
def job(email,passwd):
    I_want_money = IQ_Option(email, passwd)
    id_number=I_want_money.buy(2,"EURJPY","call",1,force_buy=False)
    print("buy-time",time.time(),id_number)
p1 = mp.Process(target=job,args=("email","passwd"))
 
p2 = mp.Process(target=job,args=("email","passwd"))
 
p3 = mp.Process(target=job,args=("email","passwd"))
 
p1.start()
p2.start()
p3.start()

p1.join()
p2.join()
p3.join()
```

![image](https://user-images.githubusercontent.com/25873606/55403340-3b480680-5588-11e9-87e5-564a56225458.png)

**Comment by jackynguyen93** [comment]
@Lu-Yi-Hsun do you have another solution. I think this solution will cause to delay on order. In my case, it takes about 6s to authentication and makes a connection.
![Capture](https://user-images.githubusercontent.com/14976555/55407493-41d67e00-5588-11e9-9157-17fbd7d59e4c.PNG)
Can we do a thread safe for api or using scope variable instead of global variable to make api doesn
t get the last response only

**Comment by Lu-Yi-Hsun** [comment]
you can start the multiprocessing wait for buy() action

**Comment by jackynguyen93** [comment]
> you can start the multiprocessing wait for buy() action

Sorry, I don't understand what you mean. An example will more clear with me

**Comment by Lu-Yi-Hsun** [comment]
you need "6s to authenticate" just first! 
use "zmq" to connect multiprocessing request what you want

**Comment by jackynguyen93** [comment]
so y

> you need "6s to authenticate" just only at first time!
> 
> use "zmq" to connect multiprocessing request what you want

So you mean we run many threads at start time and do authenticate for all threads. Then zmq will send signal to each thread and enter order in each thread ?
